sklearn/linear_model/sag_fast.pyx.tp

```diff
@@ -605,8 +626,9 @@ cdef void lagged_update{{name}}({{c_type}}* weights, {{c_type}} wscale, int xnnz
                 if reset:
                     weights[idx] *= wscale
                     if not skl_isfinite{{name}}(weights[idx]):
-                        with gil:
-                            raise_infinite_error(n_iter)
+                        # do not return now because it requires the gil
+                        status = -1
+                        break
```

Comment:
maybe you could directly return here since the function's type is `int`

Reply:
you're right, thanks!